source/CIS/cis-0.rst

```diff
@@ -0,0 +1,95 @@
+
+Abstract
+========
+
+A standardized interface for how a smart contract indicate supported smart contract standards.
+Additionally, this specification includes a list of identifiers for each standard used on the Concordium blockchain.
+
+The interface provides an entrypoint for querying the smart contract about whether it supports a given standard. The smart contract responds with either: not supported, supported or supported by some contract address.
+
+Specification
+=============
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED",  "MAY", and "OPTIONAL" in this document are to be interpreted as described in :rfc:`2119`.
+
+.. _CIS-0-standard-identifer:
+
+Standard identifer
+-------------------
+
+Each standard MUST have a unique identifier.
+An identifier is a variable-length ASCII string up to 255 characters.
+
+The identifier is serialized as: first 1 byte for the length followed this many bytes for the ASCII encoding of the identifier::
+
+  StandardIdentifier ::= (n: Byte) (id: Byteⁿ)
+
+The table below contains the current standards and associated standard identifier.
+
+Contract function
+-----------------
+
+A smart contract implementing CIS-0 MUST export the following function: :ref:`CIS-0-functions-supports`.
+
+.. _CIS-0-functions-supports:
+
+``supports``
+^^^^^^^^^^^^
+
+Query supported standards using a list of standard identifers.
+
+Parameter
+~~~~~~~~~
+
+The parameter consists of a list of standard identifers.
+
+It is serialized as: 2 bytes for the number (little endian) of standard identifers and then this number of :ref:`CIS-0-standard-identifer` identifers::
+
+  SupportsParameter ::= (n : Byte²) (ids: StandardIdentifierⁿ)
+
+Response
+~~~~~~~~
+
+The function output is a list of support results, where the order of the support results matches the order of standard identifers in the parameter.
+
+It is serialized as: 2 bytes for the number (little endian) of results (``n``) and then this number of support results (``results``).
+A support result is serialized as either: a byte with value ``0`` for "Standard is not supported", a byte with the value ``1`` for "Standard is support by this contract" or a byte with value ``2`` for "Standard is supported by using another contract address" followed by a single byte encoding the number of contract address then this number of contract addresses (``address``).
+A contract address is serialized as: first 8 bytes for the index (``index``) followed by 8 bytes for the subindex (``subindex``) both little endian::
+
+  ContractAddress ::= (index: Byte⁸) (subindex: Byte⁸)
+
+  SupportResult ::= (0 : Byte)                             // Standard is not supported
+                  | (1 : Byte)                             // Standard is supported by this contract
+                  | (2 : Byte) (n: Byte) (address: ContractAddressⁿ)  // Standard is supported by using one of these contract addresses.
```

Comment:
I think we should still consider adding more details as to what this means. We could give an example with proxies.

Reply:
I have added a note in the description of the `supports` function, let me know what you think